### PR TITLE
[SCV-17] Item Collections Missing Fields

### PR DESCRIPTION
**Pull request by quetcodesfire**
Item collections were missing the `type` and `links` fields, looking through the codebase is looked like collections coming from the `stac/search` endpoint were being returned correctly, however, returning from the `collections/"collectionId` endpoints they were being returned incorrectly. 

To fix this I brought over the pagination from `/lib/convert/granules` and added the `type` and `links` fields to `granulesResponse`.

### CODE REVIEW

File: search/lib/api/wfs.js [hunk 1]
```diff
@@ -32,8 +33,20 @@ async function getGranules (request, response) {
...
+  const nextPage = currPage + 1;
+  const newParams = { ...event.queryStringParameters } || {};
+  newParams.page_num = nextPage;
+  const nextResultsLink = generateAppUrlWithoutRelativeRoot(event, event.path, newParams);
+
   const granulesResponse = {
-    features: granules.map(gran => convert.cmrGranToFeatureGeoJSON(event, gran))
+    type: 'FeatureCollection',
+    features: granules.map(gran => convert.cmrGranToFeatureGeoJSON(event, gran)),
+    links: {
```
Comment (DrewE84) [inline search/lib/api/wfs.js:46]: I think `links` is supposed to be an array according the spec here https://github.com/radiantearth/stac-spec/blob/v0.8.1/item-spec/itemcollection-spec.md

File: search/tests/api/wfs.spec.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ describe('wfs routes', () => {
 
       expect(cmr.findGranules).toHaveBeenCalled();
       expect(convert.cmrGranToFeatureGeoJSON).toHaveBeenCalled();
-      expect(response.json).toHaveBeenCalledWith({ features: [{ response: 'okay' }] });
+      expect(response.json).toHaveBeenCalledWith({ features: [{ response: 'okay' }], "links": {"next": "http://example.com?page_num=2", "self": "http://example.com"}, "type": "FeatureCollection" });
```
Comment (DrewE84) [inline search/tests/api/wfs.spec.js:95]: To follow my comment above, I think `links` is expected to be an array.

File: search/lib/api/wfs.js (hunk 1 above)
Comment (quetcodesfire) [reply search/lib/api/wfs.js]: You're right, I've fixed this in the latest commit.

File: search/tests/api/wfs.spec.js (hunk 2 above)
Comment (quetcodesfire) [reply search/tests/api/wfs.spec.js]: Fixed in the latest commit.

**Pull request merged**